src/main/scala/zio/redis/options/zinterstore.scala

```diff
@@ -0,0 +1,17 @@
+package zio.redis.options
+
+object zinterstore {
+
+  sealed trait Key
+  object Key {
+    case class SimpleKey(key: String)                   extends Key
```

Comment:
Please make all case classes final.

Also, having in mind the comment above, you might end up without `Key`, but with 2 simple case classes.

Reply:
After reorganizing the modules into `shared`, `zset` etc. and making the case classes final in the `zset`trait I ran into warnings like:
```
[warn] /Users/manfred/src/github/zio-redis/src/main/scala/zio/redis/options/zset.scala:8:22: The outer reference in this type test cannot be checked at run time.
[warn]     final case class Open(value: String)   extends LexMinimum
```

Reply:
Oh, I hate this one. You should be able to seal it instead.

Reply:
All sealed up! I.e. replaced `case class` with `sealed case class` in `zio/redis/options/SortedSets.scala`.